src/storage/txn/scheduler.rs

```diff
@@ -1011,19 +1008,23 @@ impl<E: Engine, L: LockManager> Scheduler<E, L> {
                             reason: kvrpcpb::WriteConflictReason::PessimisticRetry,
                         },
                     )));
-                    cb(Err(e.into()));
+                    cb(Err(e.into()), false);
                 }
 
                 for f in delayed_wake_up_futures {
+                    let self2 = self1.clone();
                     self1
                         .get_sched_pool(CommandPri::High)
                         .pool
                         .spawn(async move {
                             let res = f.await;
-                            // It returns only None currently.
-                            // TODO: Handle not-none case when supporting resumable pessimistic lock
-                            // requests.
-                            assert!(res.is_none());
+                            if let Some(resumable_lock_wait_entry) = res {
+                                self2.schedule_awakened_pessimistic_locks(
+                                    None,
+                                    None,
+                                    smallvec![resumable_lock_wait_entry],
+                                );
+                            }
```

Comment:
This change should have been done in the previous PR but mistakenly missing and not covered by the tests. It's now covered by the new failpoint test.